library/core/src/escape.rs

```diff
@@ -0,0 +1,97 @@
+//! Helper code for character escaping.
+
+use crate::num::NonZeroUsize;
+use crate::ops::Range;
+
+const HEX_DIGITS: [u8; 16] = *b"0123456789abcdef";
+
+/// Escapes a byte into provided buffer; returns length of escaped
+/// representation.
+pub(super) fn escape_ascii_into(output: &mut [u8; 4], byte: u8) -> Range<u8> {
```

Comment:
nit: could you make these `pub(crate)` instead of `pub(super)`?  When I see `super` it makes me try to figure out what outside the parent module shouldn't be able to access them (like if there's a meaningful encapsulation boundary) but if it's just that they shouldn't be exported from `core` (and that using them anywhere in `core` is fine) I'd rather that be the more usual `pub(crate)` than `pub(super)`-that-happens-to-be-the-whole-crate.